src/popup/modules/UserInterface.js

```diff
@@ -532,6 +532,12 @@ export function init() {
     const applyingQrColor = AddonSettings.get("qrBackgroundColor").then((qrBackgroundColor) => {
         if (qrBackgroundColor) {
             document.body.style.backgroundColor = qrBackgroundColor;
+
+            // checks for OS dark theme and sets sets colors in text field
+            if (window.matchMedia('(prefers-color-scheme: dark)').matches) {
+                qrCodeText.style.backgroundColor = "#4a4a4f"; // Photon Grey 60 -- FF dark theme popup color
```

Comment:
Why did you move that from CSS to JS now?
IMHO a combination would be just fine…

Everything which you can define statically, put into CSS. (as it was before)
And everything which needs JS-handling, use JS…

Reply:
Moving back to CSS